client/index.js

```diff
@@ -0,0 +1,23 @@
+const express = require('express');
+const next = require('next');
+const logger = require('pino-http')()
+
+const port = parseInt(process.env.PORT, 10) || 3000;
+const isDev = process.env.NODE_ENV !== 'production';
+
+const app = next({ dev: isDev });
+const handle = app.getRequestHandler();
+
+app.prepare().then(() => {
+  const server = express();
+
+  server.use(express.json());
+  server.use(logger);
+
+  server.all('*', (req, res) => handle(req, res));
+
+  server.listen(port, err => {
+    if (err) throw err;
+    console.log(`> Ready on http://localhost:${port}`);
+  });
+});
```

Comment:
A custom server is not compatible with Vercel, did you add it because of the logger right?

Reply:
yes. happy to explore other solutions that keep vercel happy, as long as we get to have logs.

Reply:
after some prettification and adjusting vercel settings to match the new structure, it seems to be running just fine, so maybe we don't have the custom logging on vercel, but I can live with that